src/main/java/uk/ac/sanger/sccp/stan/service/register/RegisterServiceImp.java

```diff
@@ -110,12 +111,33 @@ public Map<String, Tissue> createTissues(RegisterRequest request, RegisterValida
                     donor,
                     validation.getMedium(block.getMedium()),
                     validation.getFixative(block.getFixative()),
-                    hmdmc);
+                    hmdmc, block.getSampleCollectionDate());
             tissueMap.put(tissueKey, tissueRepo.save(tissue));
         }
         return tissueMap;
     }
 
+    /**
+     * Updates any existing tissues that now have a collection date
+     * @param request specification
+     * @param validation validation result to look up tissues
+     */
+    public void updateExistingTissues(RegisterRequest request, RegisterValidation validation) {
+        List<Tissue> toUpdate = new ArrayList<>();
+        for (BlockRegisterRequest brr : request.getBlocks()) {
+            if (brr.isExistingTissue() && brr.getSampleCollectionDate()!=null) {
+                Tissue tissue = validation.getTissue(brr.getExternalIdentifier());
+                if (tissue!=null && tissue.getCollectionDate()==null) {
+                    tissue.setCollectionDate(brr.getSampleCollectionDate());
```

Comment:
@khelwood Just a doubt, why do we need to keep the collection date again in each tissue, other than keeping it in the block?

Reply:
This is the place it is stored in the database: the tissue table.